### PR TITLE
[Ready for review] Geothermal Power instead of magic solars

### PR DESCRIPTION
**Pull request by TheLordME**
## About The Pull Request
- [x] UI for the geothermal controller
- [x] way to improve poweroutput

## Why It's Good For The Game
really makes sense for solars to output 400 kW while you cant see a thing outside due to snow/night time

## Changelog
:cl:
add: Adds a geothermal power station on the western plains
add: Adds the machines for the geothermal power station
del: Removed solars
/:cl:

### CODE REVIEW

File: code/modules/power/geothermal_power.dm [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+		if(500 KILOWATTS to 1 MEGAWATTS)
+			icon_state = "controller_med"
+		if(1 MEGAWATTS to 2 MEGAWATTS)
+			icon_state = "controller_high"
+		if(2 MEGAWATTS to INFINITY)
+			icon_state = "controller_vhigh"
+		else
+			icon_state = "controller_idle"
+	if(use_power == USE_POWER_OFF)
+		icon_state = "controller_off"
```
Comment (Zandario) [inline code/modules/power/geothermal_power.dm:54]:
```suggestion
/obj/machinery/power/geothermal_controller/update_icon_state()
	switch(power_total)
		if(50 to 500 KILOWATTS)
			icon_state = "controller_low"
		if(500 KILOWATTS to 1 MEGAWATTS)
			icon_state = "controller_med"
		if(1 MEGAWATTS to 2 MEGAWATTS)
			icon_state = "controller_high"
		if(2 MEGAWATTS to INFINITY)
			icon_state = "controller_vhigh"
		else
			icon_state = "controller_idle"
	if(use_power == USE_POWER_OFF)
		icon_state = "controller_off"
	return ..()
```

File: code/modules/power/geothermal_power.dm [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+/obj/machinery/power/geothermal_collector
+	name = "Akureyri Geothermal Power Collector"
+	desc = "The power collecting part of the Akureyri Geothermal Power system, built to withstand serious heat."
+
+	icon = 'icons/obj/machines/power/geothermal.dmi'
+	icon_state = "pipe"
+
+	anchored = 1
+	density = 0
```
Comment (Zandario) [inline code/modules/power/geothermal_power.dm:83]:
```suggestion
	anchored = TRUE
	density = FALSE
```

File: code/modules/power/geothermal_power.dm [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+/obj/machinery/power/geothermal_controller
+	name = "Akureyri Geothermal Power Controller"
+	desc = "The Akureyri Geothermal Power Controller is the specialised geothermal Power solution offered to you by Frag Felix Storage."
+
+	icon = 'icons/obj/machines/power/geothermal.dmi'
+	icon_state = "controller_off"
+
+	dir = NORTH
+	anchored = 1
+	density = 1
```
Comment (Zandario) [inline code/modules/power/geothermal_power.dm:19]:
```suggestion
	anchored = TRUE
	density = TRUE
```

File: code/modules/power/geothermal_power.dm [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+			LAZYADD(collectors, col)
+
+/obj/machinery/power/geothermal_controller/examine(mob/user)
+	. = ..()
+	if(use_power == USE_POWER_OFF)
+		. += "Use a multitool to set it up."
+
+/obj/machinery/power/geothermal_controller/attackby(obj/item/W, mob/user)
+	if(W.is_multitool())
+		use_power = USE_POWER_IDLE
```
Comment (Zandario) [inline code/modules/power/geothermal_power.dm:72]:
```suggestion
		update_use_power(USE_POWER_IDLE)
```

File: code/modules/power/geothermal_power.dm [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+		else
+			icon_state = "controller_idle"
+	if(use_power == USE_POWER_OFF)
+		icon_state = "controller_off"
+	return ..()
+
+
+/obj/machinery/power/geothermal_controller/Initialize(mapload)
+	. = ..()
+	for (var/obj/machinery/power/geothermal_collector/col in range(7, src))
```
Comment (Linzolle) [inline code/modules/power/geothermal_power.dm:60]: shouldnt this be using the controller's range var instead of hardcoded to be 7?

Comment (TheLordME) [reply code/modules/power/geothermal_power.dm]: made it dependant on a component

**Review by silicons — DISMISSED** [review]

**Review by silicons — APPROVED** [review]

**Pull request merged**